### DISCUSSION TITLE
CODE chunk

### DISCUSSION THREAD

**Discussion by Liripo** [openwdl/wdl#470] [Q&A]
```wdl
workflow test {
	String workdir="/path/to/outdir"
	call test_do {
		input:
			workdir=workdir
	}
}

task test_do {
	String workdir
	command <<<
		cd ${workdir}
		echo "start" >log
		R <<CODE
			readr::write_lines(mtcars$am,"a.txt")
		CODE
		echo "end">>log
	>>>
}
```
while i run this code,the `$am` is parsed as a shell variable,Is this a bug or is it designed like this?
I am using cromwell to run wdl.

**Comment by DavyCats** [comment]
This sounds like more of a bash issue, `$am` isn't a WDL placeholder, so I don't think a WDL execution engine would be doing anything with it.

As a side note, it seems you're using WDL draft-2, I would recommend using WDL 1.0 if possible (or even 1.1, except that cromwell doesn't support it yet).

**Reply by Liripo** [reply]
I will try WDL1.1  if I have a chance

**Comment by pshapiro4broad** [comment]
This a feature of bash heredocs. If you want to prevent substitution, quoting the `$` could work, or you could use a heredoc form that doesn't do substitution: `<<'`. See https://tldp.org/LDP/abs/html/here-docs.html

**Reply by Liripo** [reply]
Thank you
